src/java.desktop/windows/native/libawt/windows/ThemeReader.cpp

```diff
@@ -32,7 +32,7 @@
 
 #include <uxtheme.h>
 
-#if defined(_MSC_VER) && _MSC_VER >= 1800
+#if defined(_MSC_VER)
 #  define ROUND_TO_INT(num)    ((int) round(num))
 #else
 #  define ROUND_TO_INT(num)    ((int) floor((num) + 0.5))
```

Comment:
If you look at when and why this was introduced (*), the "else" was not to support some other compiler - it was to support the older MS compiler. So if you don't want that, then the whole thing reduces to
#define ROUND_TO_INT(num) ((int) round(num))

.. you could even go further and eliminate the macro altogether if it makes sense - you'd have to look at the usages.

Same logic applies to the other files.


(*) https://mail.openjdk.java.net/pipermail/awt-dev/2016-March/010889.html

Reply:
Hi Phil, I simplified the code more and now use round directly.